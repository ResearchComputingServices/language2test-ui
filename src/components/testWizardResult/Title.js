import React from 'react';
import Divider from '@material-ui/core/Divider';

export default () => (
    <div className='text-center'>
        <h1>Test Results</h1>
        <Divider className='mt-3 mb-5' />
    </div>
);
