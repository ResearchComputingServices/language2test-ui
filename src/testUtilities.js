/* eslint-disable import/prefer-default-export */
/* eslint-disable prefer-const */
/* eslint-disable react/prop-types */
/* eslint-disable react/no-children-prop */
import React from 'react';
import { shallow, mount as mountComponent } from 'enzyme';
import { Router } from 'react-router-dom';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import _ from 'lodash';
import historyService from './services/HistoryService';
import { reducers } from './redux/slices';

const mockStore = configureMockStore({ reducers });

function WrapperComponent({ propsFunc, children, Component }) {
    const props = propsFunc();
    return (
        <Component
            children={children}
            {...props}
        />
    );
}

export function createTest(Component, opts = {}, callback) {
    let { props, children, mount, state } = opts;
    if (_.isNil(props)) props = {};
    if (_.isNil(children)) children = null;
    if (mount) {
        return () => {
            state = state || {};
            const mockMath = Object.create(global.Math);
            mockMath.random = () => 0.5;
            global.Math = mockMath;
            const mountedTree = mountComponent((
                <Router history={historyService.getHistory()}>
                    <Provider store={mockStore({ ...state })}>
                        <Component {...props} >
                            {children}
                        </Component>
                    </Provider>
                </Router>));
            _.isFunction(callback) && callback(mountedTree);
        };
    }
    if (_.isFunction(props)) {
        return () => {
            const shallowTree = shallow(
                <WrapperComponent
                    Component={Component}
                    propsFunc={props}
                >
                    {children}
                </WrapperComponent>,
            ).dive();
            _.isFunction(callback) && callback(shallowTree);
        };
    }
    return () => {
        const shallowTree = shallow((
            <Component {...props} >
                {children}
            </Component>
        ));
        _.isFunction(callback) && callback(shallowTree);
    };
}

export function createSnapshotTest(Component, opts = {}) {
    return createTest(Component, opts, tree => expect(toJSON(tree)).toMatchSnapshot());
}
