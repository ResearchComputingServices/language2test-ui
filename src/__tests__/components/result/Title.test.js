import Title from '../../../components/testWizardResult/Title';
import { createSnapshotTest } from '../../../testUtilities';

describe('Title', () => {
    it('should match snapshot', createSnapshotTest(Title));
});
