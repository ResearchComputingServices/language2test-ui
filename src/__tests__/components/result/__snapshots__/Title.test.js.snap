// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match snapshot 1`] = `
<div
  className="text-center"
>
  <h1>
    Test Results
  </h1>
  <WithStyles(ForwardRef(Divider))
    className="mt-3 mb-5"
  />
</div>
`;
